pkg/cmd/repo/edit/edit.go

```diff
@@ -240,6 +258,34 @@ func editRun(ctx context.Context, opts *EditOptions) error {
 		}
 	}
 
+	if hasSecurityEdits(opts.Edits) {
+		apiClient := api.NewClientFromHTTP(opts.HTTPClient)
+		repo, err := api.FetchRepository(apiClient, opts.Repository, []string{"viewerCanAdminister"})
+		if err != nil {
+			return err
+		}
+		if !repo.ViewerCanAdminister {
+			return fmt.Errorf("you do not have sufficient permissions to edit repository security and analysis features")
+		}
+
+		opts.Edits.SecurityAndAnalysis = &SecurityAndAnalysisInput{}
+		if opts.Edits.enableAdvancedSecurity != nil {
+			opts.Edits.SecurityAndAnalysis.EnableAdvancedSecurity = &SecurityAndAnalysisStatus{
+				Status: boolToStatus(*opts.Edits.enableAdvancedSecurity),
+			}
+		}
+		if opts.Edits.enableSecretScanning != nil {
+			opts.Edits.SecurityAndAnalysis.EnableSecretScanning = &SecurityAndAnalysisStatus{
+				Status: boolToStatus(*opts.Edits.enableSecretScanning),
+			}
+		}
+		if opts.Edits.enableSecretScanningPushProtection != nil {
+			opts.Edits.SecurityAndAnalysis.EnableSecretScanningPushProtection = &SecurityAndAnalysisStatus{
+				Status: boolToStatus(*opts.Edits.enableSecretScanningPushProtection),
+			}
+		}
```

Comment:
I think I'd like to see these moved into their own helper function and to be moved up into the `RunE` of `NewCmdEdit`. That way, we can unit test the transformer (i.e. the thing that turns the flag booleans into the struct format) and inject the `SecurityAndAnalysis` struct on `opts` as a dependency into `editRun` for testing purposes.

I was really confused at the start of this review as to why this looked so different from the other repo properties. I finally found my answer in the [API docs](https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#update-a-repository). I think this change will help future contributors grok this faster than I did, too.

Let me know if you have questions. I'm also happy to add a commit or PR into this branch to show you what I mean 🙂

Reply:
Sounds good to me! I added a helper function `transformSecurityAndAnalysisOpts()` in the latest commit, added a unit test for it, and moved the rest of the logic into `RunE`. Let me know if there's anything I missed/misunderstood from your comment.